androidHyperskillApp/src/main/java/org/hyperskill/app/android/study_plan/fragment/StudyPlanFragment.kt

```diff
@@ -44,8 +46,42 @@ class StudyPlanFragment :
     private var gamificationToolbarDelegate: GamificationToolbarDelegate? = null
     private var studyPlanWidgetDelegate: StudyPlanWidgetDelegate? = null
 
+    private val onForegroundObserver =
+        object : DefaultLifecycleObserver {
+            override fun onPause(owner: LifecycleOwner) {
+                super.onPause(owner)
+                wasInBackground = true
+            }
+        }
+
+    private var wasInPause = false
+    private var wasInBackground = false
+
+    override fun onPause() {
+        super.onPause()
+        wasInPause = true
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (wasInBackground) {
+            studyPlanViewModel.onNewMessage(
+                StudyPlanScreenFeature.Message.Initialize(forceUpdate = true)
+            )
+        } else if (wasInPause) {
+            studyPlanViewModel.onNewMessage(
+                StudyPlanScreenFeature.Message.StudyPlanWidgetMessage(
+                    StudyPlanWidgetFeature.Message.ReloadContentInBackground
+                )
+            )
```

Comment:
In my opinion, it's better to replace direct sending `StudyPlanWidgetFeature.Message.ReloadContentInBackground` with `StudyPlanScreenFeature.AppBecomesForegrund`, so that widget message will be sent in screen reducer.